### ISSUE TITLE
Is Overfiting issues? (train result is good ,but test result is bad)

### ISSUE THREAD

**Issue by heavenong** [AlexeyAB/darknet#166]
I train the voc dataset 。 train 3 days  。  the result is good  when trains  。  but  the result is bad when valid .  the result picture is below 

this is train result
![wx_camera_1503376029030](https://user-images.githubusercontent.com/25174797/29598644-afc6c100-87fc-11e7-8df8-684686aad6b6.jpg)

this is test result 
![wx_camera_1503461345767](https://user-images.githubusercontent.com/25174797/29598662-cedbed18-87fc-11e7-926a-00c39288ec3d.jpg)

Does anyone know what the reason is

Thanks

**Comment by AlexeyAB** [comment]
VOC dataset usually should be trained about 40 000 - 45 000 iterations.

1. Try to set this value to `float thresh = .001;` in the `detector.c`, recompile and do `darknet.exe detector recall ...` again: https://github.com/AlexeyAB/darknet/blob/4d2fefd75a57dfd6e60680eaf7408c82e15a025d/src/detector.c#L397
2. What cfg-file did you use `yolo-voc.2.0.cfg` or `yolo-voc.cfg`?
3. What values of `width, height, random, batch, subdivisions` in your cfg file?

**Comment by heavenong** [comment]
my cfg-file is yolo-voc.2.0.cfg

I did not make any changes to the file 
w=416
h=416
batch=64
subdivisions=8
random=0
my weights files has generate yolo14000.weight  

Will not be the number of times I trained enough, but if the number of training is not enough in the training process to see the parameters are also very good？

Some people say that is over fitting, there is no such possibility？

**Comment by AlexeyAB** [comment]
* To check, is there overfitting just do `darknet.exe detector recall` for: `yolo8000.weight`, `yolo10000.weight` and `yolo14000.weight` - if IoU grows, then there isn't overfitting

* Do you train VOC2007+2012, and how many images in your dataset?
You number of iterations should not be less then number of images.

**Comment by heavenong** [comment]
Hi @AlexeyAB 
 The number of images is about 30000 . 
files                         IOU
yolo1000.weight    34%
yolo3000.weight    36%
yolo6000.weight    36%
yolo8000.weight    40%
yolo12000.weight    40%
yolo14000.weight    40%


And  I am trainning the model use yolo.cfg
I have train 90000+ images   but the result also is bad  ,Can you tell me how to train the model is good

thanks

follow is i am trainning result
![tim 20170824092438](https://user-images.githubusercontent.com/25174797/29659521-2b820b8e-88f1-11e7-8e08-ca846575c137.png)

**Comment by AlexeyAB** [comment]
> And I am trainning the model use yolo.cfg

* Use `yolo-voc.2.0.cfg`, not `yolo.cfg` and not `yolo-voc.cfg`
* Set `random=1` in the `yolo-voc.2.0.cfg` and train 40 000 iterations

**Comment by heavenong** [comment]
@AlexeyAB 
OK ,   Let me try    
Thanks

**Comment by heavenong** [comment]
Hi @AlexeyAB 

Ask you a question, what is the random parameter mean? And how can I set the random to 1 on the basis of the last iteration of 14000.weight? And why use yolo. Cfg training speed than yolo-voc.2.0.cfg training speed. 
I see learning_rate = 0.0001 for the yolo-voc.2.0.cfg file and learning_rate for yolo.cfg and yolo-voc.cfg = 0.001.is it right？

Thanks

**Comment by AlexeyAB** [comment]
`random=1` mean that network will be randomly resized for each 10 iterations [288x288] - [608x608] to be more scale invariant, to detect objects with any sizes.

* yolo.cfg and yolo.2.0.cfg - networks optimized for MS COCO images dataset.
* yolo-voc.cfg and yolo-voc.2.0.cfg - networks optimized for Pascal VOC images dataset.

`yolo-voc.2.0.cfg` is Yolo v2. 
`yolo-voc.cfg` new some changed cfg - there are params: burin_in and increased initial learning rate 0.001 instead of 0.0001.
